core/src/smartcontracts/isi/block.rs

```diff
@@ -10,10 +13,43 @@ impl ValidQuery for FindAllBlocks {
     fn execute(&self, wsv: &WorldStateView) -> Result<Self::Output, query::Error> {
         let mut blocks: Vec<BlockValue> = wsv
             .blocks()
-            .map(|blk| blk.clone())
+            .map(|block| block.clone())
             .map(VersionedCommittedBlock::into_value)
             .collect();
         blocks.reverse();
         Ok(blocks)
     }
 }
+
+impl ValidQuery for FindAllBlockHeaders {
+    #[metrics(+"find_all_block_headers")]
+    fn execute(&self, wsv: &WorldStateView) -> Result<Self::Output, query::Error> {
+        let mut blocks: Vec<BlockHeaderValue> = wsv
+            .blocks()
+            .map(|block| block.clone())
+            .map(VersionedCommittedBlock::into_value)
+            .map(|block_value| block_value.header)
+            .collect();
+        blocks.reverse();
```

Comment:
Please, add some documentation why do you reverse it. I'd prefer add comment here and a doc comment for `FindAllBlockHeaders` describing the block order in the output

Reply:
Right, that makes sense.